### PR TITLE
Error codes

### PR DESCRIPTION
**Pull request by Neamar**

### CODE REVIEW

File: blueprint.md [hunk 1]
```diff
@@ -105,10 +114,14 @@ Retrieve data about the current account. This endpoint return:
 
 ## Token [/token]
 ### Retrieve frontend token [GET]
-> This endpoint can only be used with basic authentication.
+> This endpoint can only be used with `Basic` authentication.
 
 Create or retrieve a token. The token will always be the same until you call `/company/reset`.
 
+> * `401 Unauthorized`: you did not specify any credentials, or you're using a non-supported `Authorization` scheme
+> * `401 Unauthorized`: you tried to used `Bearer` authentication, but this endpoint can only be used with `Basic` authentication.
```
Comment (rricard) [inline blueprint.md]: you tried to use

File: blueprint.md [hunk 2]
```diff
@@ -144,12 +160,19 @@ Contains your company name, and the list of hydraters used on the account.
 ### Update documents [POST]
 Ping all providers for the current company, checking for new available documents.
 
+> * `401 Unauthorized`: you did not specify any credentials, or you're using a non-supported `Authorization` scheme
+> * `401 InvalidCredentials`: you did not specify a token, or your token is invalid / has been revoked.
+> * `429 Too Many Requests`: this endpoint is throttled.
```
Comment (rricard) [inline blueprint.md]: limit ?

File: blueprint.md [hunk 3]
```diff
@@ -24,14 +24,20 @@ Should have a `status` of "OK".
...
             }
 
+
 ## Batch calls [/batch{?pages}]
 ### Batch querying [GET]
 Queue multiple `GET` queries in a single request. For instance, you can call `/batch?pages=/&pages=/document_types&pages=/providers` to get one JSON object with all data, indexed by url.
 
 Status code will be 200 if all queries passed. If an error occured, the `errored` key will tell you which page failed to load.
 
+> * `401 Unauthorized`: you did not specify any credentials, or you're using a non-supported `Authorization` scheme
```
Comment (rricard) [inline blueprint.md]: you are ? => you're is for spoken language usually

File: blueprint.md [hunk 4]
```diff
@@ -267,6 +334,10 @@ Search within all availables data for documents matching specified filter.
 
 Return aggregated informations computed over the result set. The `score` key indicated document's relevance regarding query.
 
+> * `401 Unauthorized`: you did not specify any credentials, or you're using a non-supported `Authorization` scheme
+> * `401 InvalidCredentials`: you did not specify a token, or your token is invalid / has been revoked.
+> * `409 InvalidArgument`: malformed search query, with misused characters
```
Comment (rricard) [inline blueprint.md]: which ones ?

File: blueprint.md [hunk 5]
```diff
@@ -351,7 +423,15 @@ Send a new document on anyFetch.
...
 > You can't send a file and a document at the same time, you need to send the document first and then send the file on `/documents/:id/file`.
 
-+ Request
+> * `401 Unauthorized`: you did not specify any credentials, or you're using a non-supported `Authorization` scheme
+> * `401 InvalidCredentials`: you did not specify a token, or your token is invalid / has been revoked.
+> * `403 Forbidden`: document was not provided with this access token, and can't be updated.
+> * `409 InvalidArgument`: you should either specify `id` or `identifier` in your payload, not both.
+> * `409 InvalidArgument`: the `id` is not a valid id.
+> * `409 MissingParameter`: neither `id` nor `identifier` was specified
+> * `409 InvalidArgument`: you sent an `id`, but no document match. Use `identifier` to create a new document
```
Comment (rricard) [inline blueprint.md]: but no document matching

File: blueprint.md [hunk 6]
```diff
@@ -462,7 +559,7 @@ Result contains, amongst other :
                 "title": {
                     "subject": "Follow up Charly Kevers"
                 },
-                "document_type": "5252ce4ce4cfcd16f55cfa41",
+                "document_t&ype": "5252ce4ce4cfcd16f55cfa41",
```
Comment (rricard) [inline blueprint.md]: document_t&ype : artefact ?

File: blueprint.md [hunk 7]
```diff
@@ -570,6 +672,14 @@ Retrieve all raw data for `id` document.
 Also include information about hydraters (`hydratedBy`, `hydrating` and `lastHydration`).
 
 ### Get raw document [GET]
+View all data for the document.
+
+> * `401 Unauthorized`: you did not specify any credentials, or you're using a non-supported `Authorization` scheme
+> * `401 InvalidCredentials`: you did not specify a token, or your token is invalid / has been revoked.
+> * `404 ResourceNotFound`: document does not exists, or can't be accessed.
```
Comment (rricard) [inline blueprint.md]: exist

File: blueprint.md [hunk 8]
```diff
@@ -607,9 +717,16 @@ Work with the document's file.
 ### Get document file [GET]
 Retrieve the file associated with a document.
 
+> * `401 Unauthorized`: you did not specify any credentials, or you're using a non-supported `Authorization` scheme
+> * `401 InvalidCredentials`: you did not specify a token, or your token is invalid / has been revoked.
+> * `404 ResourceNotFound`: document does not exists, or can't be accessed.
```
Comment (rricard) [inline blueprint.md]: idem

File: blueprint.md [hunk 9]
```diff
@@ -666,7 +793,14 @@ Create a new user on this company. If `is_admin` is not specified, a standard us
 > You need to be an administrator to create another user.
 
 
-+ Request
+> * `401 Unauthorized`: you did not specify any credentials, or you're using a non-supported `Authorization` scheme
+> * `401 InvalidCredentials`: you did not specify a token, or your token is invalid / has been revoked.
+> * `403 Forbidden`: you are not an administrator for this account.
```
Comment (rricard) [inline blueprint.md]: of ? instead of for

File: javascripts/tools.js [hunk 10]
```diff
@@ -15,5 +13,25 @@ $(function() {
     }
   });
 });
+$('body').scrollspy({ target: '#nav' });
```
Comment (rricard) [inline javascripts/tools.js:16]: :+1: Likey jQuery

File: blueprint.md [hunk 11]
```diff
@@ -87,6 +93,9 @@ Retrieve data about the current account. This endpoint return:
 - `_url` to different part of the API (HATEOAS design). Notice the `current_user_url` key.
```
Comment (Boudedi) [inline blueprint.md]: `_url` to A different part.. ?

File: blueprint.md [hunk 12]
```diff
@@ -233,11 +266,45 @@ Only available for admin users.
...
+                    "http://image.hydrater.anyfetch.com/hydrate",
+                    "http://ocr.hydrater.anyfetch.com/hydrate",
+                    "http://eml.hydrater.anyfetch.com/hydrate"
+                ]
+            }
+
 ### Delete a subcompany [DELETE]
 Delete the subcompany, **all** its documents and **all** its users.
 
 By default, you won't be able to remove a subcompany with subsubcompanies. If it is really what you want to do, add `?force=1`.
```
Comment (Boudedi) [inline blueprint.md]: By default, you are not allowed to remove a subcompany with subsubcompanies. To force the subcompany removal, add `?force=1`.

File: blueprint.md [hunk 13]
```diff
@@ -267,6 +334,10 @@ Search within all availables data for documents matching specified filter.
 
 Return aggregated informations computed over the result set. The `score` key indicated document's relevance regarding query.
```
Comment (Boudedi) [inline blueprint.md]: Weird formulation : "return aggregated..." -> can you use a verb ?

Comment (Boudedi) [inline blueprint.md]: The `score` key indicates the document's relevance for this specific query.

File: blueprint.md [hunk 14]
```diff
@@ -267,6 +334,10 @@ Search within all availables data for documents matching specified filter.
...
 
+> * `401 Unauthorized`: you did not specify any credentials, or you are using a non-supported `Authorization` scheme
+> * `401 InvalidCredentials`: you did not specify a token, or your token is invalid / has been revoked.
+> * `409 InvalidArgument`: malformed search query, with misused characters (such as `+ - && || ! ( ) { } [ ] ^ \" ~ * ? : \\`)
+
 + Parameters
     + search (optional, string, `john smith`) ... Search query, probably the most important parameter for this query
```
Comment (Boudedi) [inline blueprint.md:358]: probably the most important parameter for this query -> isn't this obvious ?

File: blueprint.md [hunk 15]
```diff
@@ -351,7 +423,15 @@ Send a new document on anyFetch.
 
 > You can't send a file and a document at the same time, you need to send the document first and then send the file on `/documents/:id/file`.
 
-+ Request
+> * `401 Unauthorized`: you did not specify any credentials, or you are using a non-supported `Authorization` scheme
+> * `401 InvalidCredentials`: you did not specify a token, or your token is invalid / has been revoked.
+> * `403 Forbidden`: document was not provided with this access token, and can't be updated.
+> * `409 InvalidArgument`: you should either specify `id` or `identifier` in your payload, not both.
+> * `409 InvalidArgument`: the `id` is not a valid id.
```
Comment (Boudedi) [inline blueprint.md]: why "the" ? -> `id` is not a valid id.
why "you" ? -> for an error comment, the use of "you" seems to be a little too intrusive, don't you think ?

File: blueprint.md [hunk 16]
```diff
@@ -105,10 +114,14 @@ Retrieve data about the current account. This endpoint return:
 
 ## Token [/token]
 ### Retrieve frontend token [GET]
-> This endpoint can only be used with basic authentication.
+> This endpoint can only be used with `Basic` authentication.
 
 Create or retrieve a token. The token will always be the same until you call `/company/reset`.
 
+> * `401 Unauthorized`: you did not specify any credentials, or you are using a non-supported `Authorization` scheme
```
Comment (pmdartus) [inline blueprint.md:121]: 2 times => `401 unauthorized` error, you can merge them

Comment (Neamar) [inline blueprint.md:121]: No, i'd rather keep them separate. The same error can appear for differents reasons

**Pull request merged**